### ISSUE TITLE
Scope issue about mlx.core.uint32

### ISSUE THREAD

**Issue by Redmept1on** [ml-explore/mlx#496]
**Describe the bug**
numpy.uint32 supports a range of 0 to 4294967295, but the range represented by mlx.core.uint32 seems to be only 0 to 2147483647, which is the range represented by uint16. When 2147483648 appears in the array, the program crashes.

**To Reproduce**

Include code snippet
```python
import mlx.core as mx
import numpy as np

x = np.array([0,4294967295], dtype=np.uint32)
print("numpy:", np.isposinf(x))

x = mx.array([0,2147483647], dtype=mx.uint32)
result = mx.isposinf(x)
print("mlx1:", result)

x = mx.array([0,2147483648], dtype=mx.uint32)
result = mx.isposinf(x)
print("mlx2:", result)
```
![image](https://github.com/ml-explore/mlx/assets/84373897/d4e2e700-6c44-453a-bbb4-a9975bc9b9dc)

**Expected behavior**
mlx2: array([False, False], dtype=bool)

**Desktop (please complete the following information):**
 - OS Version: MacOS 14.2.1
 - Version 0.10.0

**Comment by a1eaiactaest** [comment]
Can confirm, reproducible.

**Comment by awni** [comment]
Thanks for finding fix in #501
